builder/builder-context.c

```diff
@@ -248,6 +252,24 @@ builder_context_get_download_dir (BuilderContext *self)
   return self->download_dir;
 }
 
+GPtrArray *
+builder_context_get_sources_dirs (BuilderContext *self)
+{
+  return self->sources_dirs;
+}
+
+void
+builder_context_set_sources_dirs (BuilderContext *self,
+                                  GPtrArray      *sources_dirs)
+{
+  if (self->sources_dirs != sources_dirs)
```

Comment:
This will most certainly never be the case, unless the very same pointer is passed, which can't happen in this branch. I am not sure it's worth checking for it.